service/controller/resource/ipam/virtualnetwork_collector.go

```diff
@@ -79,7 +82,7 @@ func NewVirtualNetworkCollector(config VirtualNetworkCollectorConfig) (*VirtualN
 func (c *VirtualNetworkCollector) Collect(ctx context.Context, _ interface{}) ([]net.IPNet, error) {
 	var err error
 	var mutex sync.Mutex
-	var reservedVirtualNetworks []net.IPNet
+	reservedVirtualNetworks := c.reservedCIDRs
```

Comment:
Small nitty detail :) 

Maybe better copy `c.reservedCIDRs` to `reservedVirtualNetworks`, so we don't accidentally modify `c.reservedCIDRs` via `reservedVirtualNetworks`, as it is the same reference this way (which would cause to treat some range as reserved as long as azure-operator pod is running).

Something like:
```suggestion
	reservedVirtualNetworks := append([]whateverTypeHere{}, c.reservedCIDRs...)
```

Makes sense?

Reply:
yep, done